src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -858,12 +899,48 @@ impl LocalFingerprint {
         match self {
             LocalFingerprint::Precalculated(..) => "precalculated",
             LocalFingerprint::CheckDepInfo { .. } => "dep-info",
+            LocalFingerprint::CheckDepInfoChecksums { .. } => "dep-info-checksums",
             LocalFingerprint::RerunIfChanged { .. } => "rerun-if-changed",
             LocalFingerprint::RerunIfEnvChanged { .. } => "rerun-if-env-changed",
         }
     }
 }
 
+fn dep_info_shared(
+    pkg_root: &Path,
+    target_root: &Path,
+    dep_info: &PathBuf,
+    cargo_exe: &Path,
+    gctx: &GlobalContext,
+) -> Result<Either<StaleItem, RustcDepInfo>, anyhow::Error> {
```

Comment:
Not thrilled with using `Either`, I feel like its obscuring what this is doing.

Reply:
For both of these, the problems go away I think if we merge the `LocalFingerprint` variants

Reply:
Function was removed as part of d422f64ca56a7b6dfab5c8e5d4bbb06cccbbe5b3